keyboards/handwired/kbod/keymaps/default/keymap.c

```diff
@@ -53,25 +58,24 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
   ),
 };
 
-const uint16_t PROGMEM fn_actions[] = {
-  [0]  = ACTION_FUNCTION(0),
-  [1]  = ACTION_FUNCTION(1),
-};
-
-
-void action_function(keyrecord_t *record, uint8_t id, uint8_t opt) {
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
   static uint8_t shift_esc_mask;
   static uint8_t alt_mask;
-  switch (id) {
-    case 0:
+
+  switch (keycode) {
+    case SFT_ESC:
       shift_esc_mask = MODS_PRESSED(SHIFT);
       SET_WHETHER(shift_esc_mask, KC_ESC, KC_GRAVE);
+
       break;
```

Comment:
```suggestion
      return false;
```
I suggest changing switches in `process_record_user` functions in these files the following way:
1. Replace `break`s with `return false` or `return true`, depending on the case (custom keycodes like this should usually return false to prevent further processing by QMK).
2. Add a default case that passes the keycode on:
```c
default:
  return true;
``` 
3. Add empty lines _after_ the `return false`/`return true` rather than before.
4. Remove the `return true` at the end of the function as it's no longer needed.
5. Make the indentation consistent: either indent all of the `case`s compared to `switch`, or don't indent any of them (preferred).

This way, the structure of all cases is the same, it's easier to maintain them as the changes are localized, and it's harder to make a mistake like using `return true`/`break` for a custom keycode that shouldn't be passed on.

```c
bool process_keycode_user(uint16_t keycode, keyrecord_t *record) {
  switch (keycode) {
  case MY_KEY1:
    // Custom keycode
    return false;

  case MY_KEY2:
    // Custom keycode
    return false;

  case KC_ESC:
    // Existing keycode, extra behavior
    return true;

  default:
    return true;
  }
}
```

Reply:
1. Yep, all the `SFT_ESC`s should in fact be returning false. Will do.
 2. I would expect that if the keycode doesn't match any of the cases, it will fall through to the final `return true`. Although, perhaps it could be easier for newbies/non-programmers to understand the `default` case, in which case it should be required for all `process_record_user()`s. @drashna ?
 3. That does help visually separate the `case`s, less so when you place an empty line before the `return` as I prefer to do to also improve readability. If you can find a majority style I'll follow it 😄 
 4. As 2.
 5. I think anything inside a set of braces *must* be indented one level further. Consider a situation where you need to use them in a `case`, it doesn't look nice to me at all:

```c
switch (thing) {
case 0: {
  blah();

  break;
}
}
```

End of the day, I don't really want this PR to turn into "prettifying everyone's keymaps for them" :)

Reply:
2. and 4. It's a good idea in general for all `switch`es to have a `default` case if all possible values are not covered by `case`s. Some compilers issue a warning if this is not so. Relying on skipping the switch (i.e. no matching case) or case fall-through is generally not a good idea.

3. Matches for `\n\n *break;\n(?!\n)` (empty line before): 48; matches for `(?<!\n)\n *break;\n\n` (empty line after): 519. Some of those are probably inside loops rather than switches, but most of them are in switches, so I think the numbers are good. I used `break` instead of `return` to reduce the number of false positives since `break` is more common in non-keycode switches. Here are the results with `return (true|false)`: before 78, after 102.

5. It's personal preference, so either one is fine as long as it's consistent 🙂 `case`s are actually jump labels, and labels aren't part of the executing code, so they traditionally aren't indented. For that reason (not being part of the program), nothing else should go on the same line as the case. In your example:
```c
switch (thing) {
case 0:
  {
    blah();
    break;
  }
}
```
This is the same as opening a block in the middle of a function, so it makes sense.
```c
void f(void) {
  char c = '1';
  {
    char c = '2';
    putchar(c);
  }
  putchar(c);
}
```

Reply:
Hm. What about when the switch is inside the `if(record->event.pressed)`?

https://github.com/qmk/qmk_firmware/pull/4935/files#diff-8989f35cd79e650ad6b39fd66d04025f

Moving the keydown check into each case seems way too verbose, and having two `return true`s seems...weird.

Reply:
That's a bit of an oddball, but I'd still say `default: return true;` makes sense in that case 😃 If you really don't like it, you can do `default: break;`, but I'd say:
1. Definitely have a `default` case.
2. Use another `return true` regardless of how it looks. Think about it from the perspective of what's least likely to cause a mistake when someone else comes along and edits that code. If you use a `break`, they might overlook it and inadvertently add code below the `if` that will be executed whenever one of the built-in keycodes is pressed/released. With `return true` this cannot happen.

Reply:
It's the only one (that this PR changes) so I've just gone with the rest of them and added the keydown check to each case. This say if you need to check keydown *and* keyup you don't have two switches, which I think is probably worse than having keydown/up checks per keycode 😄

Reply:
That works 😄 Now, what about the empty lines? Are you going to keep them above the `return`s, or are you going to move them below? Existing usage is in favor of having them below (see a couple of comments back).

Reply:
@fauxpark Can't say I'm a big fan of the first case having an empty line before `return` and the `default` one not 😄 I like PEP 8's approach for this one: vertical whitespace should be used sparingly, and only to indicate logical sections or breaks in normal execution flow (e.g. after `break`, `return`, `continue`). I wouldn't put blank lines between blocks by default, but rather only if it makes logical sense (i.e. if they perform logically distinct roles within the function). It's a bit laxer in C (especially since we're using OTBS), but I still wouldn't go wild with too much vertical whitespace.

@pelrun Will do.

Reply:
To me it makes sense to not have an empty line after `default` because the `return` is one level down. You wouldn't do this, for example:

```c
if (condition) {

  doSomething();
}
```

That is wasting vertical whitespace. Similarly, I don't believe you should have any vertical whitespace after the `break`, `return` or `continue` because in almost all situations the next line will be a `}` one level up. In a `case`, I consider the braces implied (though, as you said, they're just glorified gotos, not actual block statements).

IMO, empty lines between block statements, and between keywords and "regular" statements like an assignment and a `return` helps with readability, and makes the code look much neater and less crowded.

Reply:
I agree with most of that, but it's exactly because of the lack of the closing brace that I think the blank line after `return` is required. It's there to separate cases from each other (think logical blocks).
And I can appreciate how a blank line before a `return` can improve readability in a function with several statements one after the other (and not just before `return`, but in between some of the statements as well if there are lots of them), but I don't think that is the case here.
```c
case SFT_ESC:
  if (record->event.pressed) {
    // ...
  }
  return false;
```
That is already well separated by the braces, as is. I don't think any additional clarity is gained from adding an extra newline. Even with an `else` it wouldn't be needed — OTBS is working to our advantage here.

Reply:
I still think the `return` is logically separate from the keydown check - it will happen regardless of whether the contents of the if statement are called, and doesn't have all that much to do with the actual handling of the keycode other than telling QMK whether or not to continue trying to process it.